src/borg/locking.py

```diff
@@ -206,6 +244,9 @@ def modify(self, key, op):
         except KeyError:
             elements = set()
         if op == ADD:
+            # If we are adding exclusive lock, that means all other entries here are stale, get rid of them.
+            if key == EXCLUSIVE:
+                elements = set()
```

Comment:
have to check this again.

but iirc: an exclusive lock can only be obtained, if all shared locks are gone.